internal/dispatch/keys/hasher_ristretto.go

```diff
@@ -0,0 +1,72 @@
+//go:build !wasm
+// +build !wasm
+
+package keys
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dgraph-io/ristretto/z"
+
+	"github.com/cespare/xxhash/v2"
+)
+
+// dispatchCacheKeyHash computres a DispatchCheckKey for the given prefix and any hashable values.
+func dispatchCacheKeyHash(prefix cachePrefix, atRevision string, args ...hashableValue) DispatchCacheKey {
+	totalEstimatedLength := len(string(prefix)) + len(atRevision) + 1 // prefix, revision and `/`
+	for _, arg := range args {
+		totalEstimatedLength += arg.EstimatedLength() + 1 // +1 for the @
+	}
+
+	hasher := newDispatchCacheKeyHasher(prefix, totalEstimatedLength)
+
+	for _, arg := range args {
+		arg.AppendToHash(hasher)
+		hasher.WriteString("@")
+	}
+
+	hasher.WriteString(atRevision)
+	return hasher.BuildKey()
+}
+
+type dispatchCacheKeyHasher struct {
+	stableHasher    *xxhash.Digest
+	sb              strings.Builder
```

Comment:
Why are you tracking these all instead of using the previous hash value as a seed and making it resumable. We're still going to end up allocating a big chunk of memory here that we don't need to.

Reply:
Nevermind, I see the next commit now.